### PR TITLE
[PC-1518] Allow client to disable path normalization

### PR DESCRIPTION
**Pull request by froodian**

### CODE REVIEW

**Review by zachmccormick — APPROVED** [review]

**Comment by zachmccormick** [comment]
this is here because we had webhooks to URLs with funky utf-8 characters in them?

**Comment by froodian** [comment]
not utf-8 but urls containing url-encoded characters - it was justeat's connected content urls specifically that exposed this - https://brazetechnology.slack.com/archives/G9WCC9NQ5/p1570634478002600 - or said differently, it's a root-cause fix to https://github.com/Appboy/platform/pull/21936

**Comment by froodian** [comment]
for more context, here's the comment I'm leaving in my pull to the upstream repo:

> See the test for context - basically I have a URL like `http://example.com/encoded/Y%2BY%2FY%3D/stuff`, but when I use fasthttp, due to path normalization the actual request that gets made is to `http://example.com/encoded/Y+Y/Y=/stuff`.  On some servers (e.g. fasthttp's server), these URLs are semantically equivalent, but on some servers they are not.  https://stackoverflow.com/questions/1957115/is-a-slash-equivalent-to-an-encoded-slash-2f-in-the-path-portion-of-a has some more context on the server-side situation as well.

File: client_test.go [hunk 1]
```diff
@@ -751,6 +751,57 @@ func TestClientDoTimeoutDisableNormalizing(t *testing.T) {
...
+	var req Request
+	req.SetRequestURI(urlWithEncodedPath)
+	var resp Response
+	for i := 0; i < 5; i++ {
+		if err := c.DoTimeout(&req, &resp, time.Second); err != nil {
+			t.Fatalf("unexpected error: %s", err)
+		}
+		hv := resp.Header.Peek("received-uri")
+		if string(hv) != urlWithEncodedPath {
+			t.Fatalf("request uri was normalized: %q. Expecting %q", hv, urlWithEncodedPath)
```
Comment (chrishaines) [inline client_test.go:791]: This isn't strictly true but this could guard against other regressions so I think it's fine.

**Review by chrishaines — APPROVED** [review]

**Review by jonhyman — APPROVED** [review]

**Comment by froodian** [comment]
they merged my pull into master so I'm just going to close this and use theirs

**Pull request closed without merging**